test/operations.jl

```diff
@@ -22,4 +22,7 @@ end
     reg = rand_state(4)
     @test all(state(reg + (-reg)).==0)
     @test all(state(reg*2 - reg/0.5) .== 0)
+
+    reg = rand_state(3)
+    @test reg'*reg ≈ 1
```

Comment:
Here is the fall back test for the bug. Which is mentioned in issue https://github.com/QuantumBFS/Yao.jl/issues/176

Reply:
Also, avoiding comparing `reg` and `reg'` directly is nessesary.